app/policies/app_group_policy.rb

```diff
@@ -0,0 +1,57 @@
+class AppGroupPolicy < ApplicationPolicy
```

Comment:
lots of logic duplication, can we extract the duplicated logic into private method of this class?